src/main/scala/cromwell/engine/SymbolTable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding.{FullyQualifiedName, Call, WdlBinding, WdlValue}
+import scala.collection.mutable
+
+class SymbolTableEntry(val scope: String, val name: String, val iteration: Int, var wdlValue: Option[WdlValue], val wdlType: WdlType, input: Boolean) {
+  def isInput: Boolean = input
+
+  def isOutput: Boolean = !isInput
+}
+
+class SymbolTable(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]) {
+  val table = mutable.Set[SymbolTableEntry]()
```

Comment:
As an example you could use akka persistent actors so if the flip gets switched back on the actor self-populates with its last state